pkg/cloudclient/aws/aws.go

```diff
@@ -51,40 +61,18 @@ func (c *Client) VerifyDns(ctx context.Context, vpcID string) *output.Output {
 }
 
 // NewClient creates a new CloudClient for use with AWS.
-func NewClient(ctx context.Context, logger ocmlog.Logger, creds interface{}, region, instanceType string, tags map[string]string) (client *Client, err error) {
-	switch c := creds.(type) {
-	case awscredsv1.Credentials:
-		var value awscredsv1.Value
-		if value, err = c.Get(); err == nil {
-			client, err = newClient(
-				ctx,
-				logger,
-				value.AccessKeyID,
-				value.SecretAccessKey,
-				value.SessionToken,
-				region,
-				instanceType,
-				tags,
-			)
-		}
-	case awscredsv2.StaticCredentialsProvider:
-		client, err = newClient(
-			ctx,
-			logger,
-			c.Value.AccessKeyID,
-			c.Value.SecretAccessKey,
-			c.Value.SessionToken,
-			region,
-			instanceType,
-			tags,
-		)
-	default:
-		err = fmt.Errorf("unsupported credentials type %T", c)
+func NewClient(input *ClientInput) (client *Client, err error) {
+	client, err = newClient(input)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create AWS client: %w", err)
 	}
+	return
+}
 
+func GetEc2ClientFromInput(input *ClientInput) (ec2.Client, error) {
+	ec2Client, err := getEc2ClientFromInput(*input)
 	if err != nil {
-		return nil, fmt.Errorf("Unable to create AWS client: %w", err)
+		return ec2Client, fmt.Errorf("unable to create EC2 Client: %w", err)
```

Comment:
return (nil, err) if there's a problem with getting. I think this is a general convention

Reply:
Unfortunately we can't return nil for ec2client. We get error: "Cannot use 'nil' as the type ec2.Client".  Do you know if there;s a workaround for that?

Reply:
Hey @ritmun 

you can return `*ec2.Client` instead in the function. 
Also, I think we can omit the creation of ec2client in the beginning as `ec2.NewFromConfig(cfg)` can create one.

(e.g 
```go
func getEc2ClientFromInput(input ClientInput) (*ec2.Client, error) {
	//Get config from profile if provided
	if input.Profile != "" {
		cfg, err := config.LoadDefaultConfig(input.Ctx,
			config.WithSharedConfigProfile(input.Profile),
			config.WithRegion(input.Region),
		)

		return ec2.NewFromConfig(cfg), err
	}

	cfg, err := config.LoadDefaultConfig(input.Ctx,
		config.WithRegion(input.Region),
		config.WithCredentialsProvider(credentials.StaticCredentialsProvider{
			Value: aws.Credentials{
				AccessKeyID: input.AccessKeyId, SecretAccessKey: input.SecretAccessKey,
				SessionToken: input.SessionToken,
			},
		}),
	)

	return ec2.NewFromConfig(cfg), err
}
```
)

Reply:
This looks good. Updated.